packages/amplify-category-hosting/lib/S3AndCloudFront/helpers/file-uploader.js

```diff
@@ -19,13 +19,14 @@ async function run(context, distributionDirPath) {
 
   const hasCloudFront = !!context?.exeInfo?.template?.Resources?.CloudFrontDistribution;
 
+  const filesToUploadFirst = fileList.filter(filePath => !filePath.includes('index.html'));
+  const filesToUploadLast = fileList.filter(filePath => filePath.includes('index.html'));
 
-  await Promise.all(fileList.map(async filePath => {
-    if(!filePath.includes('index.html')) {
-      return await uploadFileTasks.push(() => uploadFile(s3Client, hostingBucketName, distributionDirPath, filePath, hasCloudFront));
-  }}));
+  await Promise.all(filesToUploadFirst.map(async filePath => {
+      return uploadFileTasks.push(() => uploadFile(s3Client, hostingBucketName, distributionDirPath, filePath, hasCloudFront));
+  }));
```

Comment:
I missed this the first time, but please move this inside the try/catch after the spinner starts so customers can see something is uploading

Reply:
I think it doesn't matter for the user because `sequential` runs after the `spinner` and we have to push to the `uploadFileTasks` first

Reply:
I did not notice that `sequential` before and that we're accumulating deferred `() => uploadFile` callbacks. Sorry about that.

If actual upload happens in `sequential` and we'll never parallelize it then perhaps we should just make sure `fileList` has elements in right order.

Reply:
Consider just concatenating `filesToUploadFirst.concat(filesToUploadLast)` and feeding this to logic that was here before. We don't need two mappings nor Promise.all.

Reply:
That's a good point. I just noticed `sequential` when making the previous comment 😂 

Now I wonder if `sequential` is necessary. What's the purpose of `sequential`? Would it make the load too slow?

If we want to control the concurrency, would it be better to use [promise-pool](https://www.npmjs.com/package/@supercharge/promise-pool) instead?